cpp/src/arrow/compute/kernels/sort.h

```diff
@@ -0,0 +1,59 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+#pragma once
+
+#include <memory>
+
+#include "arrow/compute/kernel.h"
+#include "arrow/status.h"
+#include "arrow/util/visibility.h"
+
+namespace arrow {
+
+class Array;
+
+namespace compute {
+
+class FunctionContext;
+
+ARROW_EXPORT
+Status Sort(FunctionContext* ctx, const Array& values, std::shared_ptr<Array>* offsets);
+
+ARROW_EXPORT
+Status Sort(FunctionContext* ctx, const Datum& values, Datum* offsets);
+
+class ARROW_EXPORT SortKernel : public UnaryKernel {
```

Comment:
Please add doc comments for each public class and function. In particular, give an example (see [Take](https://github.com/apache/arrow/blob/da752fd/cpp/src/arrow/compute/kernels/take.h#L38-L45)) and note that the indices of nulls will be stably partitioned to the end of the output.